packages/component/src/Dictation.js

```diff
@@ -28,48 +29,69 @@ class Dictation extends React.Component {
       submitSendBox
     } = this.props;
 
-    setDictateInterims([]);
-    setDictateState(IDLE);
-    stopDictate();
+    if (dictateState === DICTATING || dictateState === STARTING) {
+      setDictateInterims([]);
+      setDictateState(IDLE);
+      stopDictate();
 
-    if (transcript) {
-      setSendBox(transcript);
-      submitSendBox('speech');
-      startSpeakingActivity();
+      if (transcript) {
+        setSendBox(transcript);
+        submitSendBox('speech');
+        startSpeakingActivity();
+      }
     }
   }
 
   handleDictating({ results = [] }) {
-    const { setDictateInterims, setDictateState, setSendBox } = this.props;
+    const { dictateState, setDictateInterims, setDictateState, setSendBox } = this.props;
 
-    const interims = results.map(({ transcript }) => transcript);
+    if (dictateState === DICTATING || dictateState === STARTING) {
+      const interims = results.map(({ transcript }) => transcript);
 
-    setDictateInterims(interims);
-    setDictateState(DICTATING);
+      setDictateInterims(interims);
+      setDictateState(DICTATING);
+      // TODO: Send typing
+      // sendTyping();
 
-    // This is for two purposes:
-    // 1. Set send box will also trigger send typing
-    // 2. If the user cancelled out, the interim result will be in the send box so the user can update it before send
-    setSendBox(interims.join(' '));
+      // // This is for two purposes:
+      // // 1. Set send box will also trigger send typing
+      // // 2. If the user cancelled out, the interim result will be in the send box so the user can update it before send
+      // setSendBox(interims.join(' '));
```

Comment:
is this supposed to be commented out?

Reply:
Let me finish the "speech test harness" PR and will add the implementation + test here.

Reply:
Fixed.